python/cuml/solvers/cd.pyx

```diff
@@ -195,49 +197,27 @@ class CD(Base):
            Dense vector (floats or doubles) of shape (n_samples, 1)
         """
 
-        cdef uintptr_t X_ptr
-        if (isinstance(X, cudf.DataFrame)):
-            self.gdf_datatype = np.dtype(X[X.columns[0]]._column.dtype)
-            X_m = X.as_gpu_matrix(order='F')
-            self.n_rows = len(X)
-            self.n_cols = len(X._cols)
-
-        elif (isinstance(X, np.ndarray)):
-            self.gdf_datatype = X.dtype
-            X_m = cuda.to_device(np.array(X, order='F'))
-            self.n_rows = X.shape[0]
-            self.n_cols = X.shape[1]
-
-        else:
-            msg = "X matrix must be a cuDF dataframe or Numpy ndarray"
-            raise TypeError(msg)
-
-        X_ptr = self._get_dev_array_ptr(X_m)
+        cdef uintptr_t X_ptr, y_ptr
+        X_m, X_ptr, n_rows, self.n_cols, self.dtype = \
+            input_to_array(X)
 
-        cdef uintptr_t y_ptr
-        if (isinstance(y, cudf.Series)):
-            y_ptr = self._get_cudf_column_ptr(y)
-        elif (isinstance(y, np.ndarray)):
-            y_m = cuda.to_device(y)
-            y_ptr = self._get_dev_array_ptr(y_m)
-        else:
-            msg = "y vector must be a cuDF series or Numpy ndarray"
-            raise TypeError(msg)
+        y_m, y_ptr, _, _, _ = \
+            input_to_array(y)
 
         self.n_alpha = 1
 
         self.coef_ = cudf.Series(np.zeros(self.n_cols,
-                                          dtype=self.gdf_datatype))
-        cdef uintptr_t coef_ptr = self._get_cudf_column_ptr(self.coef_)
+                                          dtype=self.dtype))
+        cdef uintptr_t coef_ptr = get_cudf_column_ptr(self.coef_)
 
         cdef float c_intercept1
         cdef double c_intercept2
         cdef cumlHandle* handle_ = <cumlHandle*><size_t>self.handle.getHandle()
 
-        if self.gdf_datatype.type == np.float32:
+        if self.dtype.type == np.float32:
```

Comment:
Just for my understanding... why can't we do `self.dtype == np.float32`? why the extra .type?

Reply:
I think it is not necessary and we should be able to do `self.dtype == np.float32` indeed, one of those small things I didn't try actually and just followed pre existing code :P. Will check it